tests/Game.class.test.js

```diff
@@ -0,0 +1,965 @@
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+    });
+
+    it('should be equal to a given initial state before start', () => {
+      const initialState = [
+        [2, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 8, 0],
+        [0, 0, 0, 16],
+      ];
+
+      const game2048 = new Game(copyState(initialState));
+
+      expect(game2048.getState()).toEqual(initialState);
+    });
+  });
+
+  describe('after start', () => {
+    it('should have a state with two cells with values 2 or 4', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(count(game2048)).toBe(2);
+      expect(sum(game2048)).toBeGreaterThanOrEqual(4);
+      expect(sum(game2048)).toBeLessThanOrEqual(8);
+    });
+
+    it('should add two cells with values 2 or 4 to a custom state', () => {
+      const game2048 = new Game([
+        [2, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 8, 0],
+        [0, 0, 0, 16],
+      ]);
+
+      game2048.start();
+
+      expect(count(game2048)).toBe(6);
+      expect(sum(game2048)).toBeGreaterThanOrEqual(34);
+      expect(sum(game2048)).toBeLessThanOrEqual(38);
+    });
+
+    it('should generate 4 less often than 2', () => {
+      let totalSum = 0;
+      const times = 50;
+
+      for (let i = 0; i < times; i++) {
+        const game2048 = new Game();
+
+        game2048.start();
+
+        totalSum += sum(game2048);
+      }
+
+      expect(totalSum).toBeGreaterThan(4 * times);
+      expect(totalSum).toBeLessThan(6 * times);
+    });
+
+    it('should generate the board with cells in random positions', () => {
+      const positions = new Set();
+
+      for (let interation = 0; interation < 50; interation++) {
+        const game2048 = new Game();
+
+        game2048.start();
+
+        const [first, second] = game2048.getState().flat().flatMap((n, i) => n > 0 ? [i] : []);
+
+        positions.add(first);
+        positions.add(second);
+      }
+
+      expect(positions.size).toBeGreaterThan(12);
+    });
+  });
+
+  describe('moveLeft', () => {
+    it('should not move the board if the game is not started', () => {
+      const initialState = [
+        [8, 0, 0, 0],
+        [0, 16, 0, 0],
+        [0, 0, 32, 0],
+        [0, 0, 0, 64],
+      ];
+
+      const game2048 = new Game(copyState(initialState));
+
+      game2048.moveLeft();
+
+      expect(game2048.getState()).toEqual(initialState);
+    });
+
+    it('should move all non-zero cells to the left', () => {
+      const initialState = [
+        [8, 0, 0, 0],
+        [0, 16, 0, 0],
+        [0, 0, 32, 0],
+        [0, 0, 0, 64],
+      ];
+
+      const game2048 = new Game(initialState);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).not.toBe(0);
+      expect(state[1][0]).not.toBe(0);
+      expect(state[2][0]).not.toBe(0);
+      expect(state[3][0]).not.toBe(0);
```

Comment:
The test case 'should move all non-zero cells to the left' doesn't start the game before moving left. This could lead to a false positive if the game implementation allows moves without starting the game.